### PR TITLE
Feat/deposit tokens button

### PR DESCRIPTION
**Pull request by ChrisChan888**
1. Create a button called "Approve spending of DAI by the pool"
2. This button function call: ask the user to approve spending of all DAI.
3. Create a button called "Deposit 10 DAI"
4. Deposit 10 DAI to pool

### CODE REVIEW

**Comment by whisidev** [comment]
preview url: https://pr-11.d2eep50fel0kg9.amplifyapp.com/

File: src/components/DepositButton.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (signer) {
+      PoolInstance.get()
+        ?.deposit(
+          TOKENS[props.symbol].getCurrentAddress(),
+          '0x' + (10 ** TOKENS[props.symbol].decimals * 10).toString(16),
+          await signer.getAddress(),
+          // deadline is very large number
+          '0x' + (10 ** 18 * 10).toString(16),
+        )
+        .catch(() => console.log('Not approved yet'))
```
Comment (whisidev) [inline src/components/DepositButton.tsx:32]: the error caught here might not be an approval error. could be insufficient gas, deadline, etc. We can `console.error(err)` here inside catch.

Can we create a separate function to check whether the user has approved yet? namely, call the ERC20 `allowance` function to see if the pool can spend the user's token.

File: src/config/contracts/pool.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { Pool } from '../../../types/ethers-contracts'
+import { PoolABI } from '../../constants/abis'
+import { getContractWithSigner } from '../../utils/contract'
+import { getCurrentNetworkId } from '../../utils/network'
+import { ChainId } from '../networks'
+export class PoolClass {
```
Comment (whisidev) [inline src/config/contracts/pool.ts:6]: can name it `Pool` instead of `PoolClass`

File: src/components/ApprovalButton.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}
+export const ApprovalButton: React.FC<ApprovalButtonProps> = (
+  props: ApprovalButtonProps,
+) => {
+  const handleClick = () => {
+    TOKENS[props.symbol]
+      .get()
+      ?.approve(
+        PoolInstance.getCurrentAddress(),
+        '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff',
```
Comment (whisidev) [inline src/components/ApprovalButton.tsx:25]: replace with `constants.MaxUint256` for readability

`import { constants } from 'ethers'`

File: src/components/DepositButton.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+export const DepositButton: React.FC<DepositButtonProps> = (
+  props: DepositButtonProps,
+) => {
+  const handleClick = async () => {
+    const signer = getSigner()
+    if (signer) {
+      PoolInstance.get()
+        ?.deposit(
+          TOKENS[props.symbol].getCurrentAddress(),
+          '0x' + (10 ** TOKENS[props.symbol].decimals * 10).toString(16),
```
Comment (whisidev) [inline src/components/DepositButton.tsx:27]: it's more standard (and less error-prone) to create a BN instance using its method than to manually concatenate a hex string.

`BigNumber.from(10).mul(BigNumber.from(10).pow(TOKENS[props.symbol].decimals))`

Comment (whisidev) [reply src/components/DepositButton.tsx:27]: in general, we can leverge 'ethers.js' whenever possible. Their API provides a lot of useful functions.

Comment (ChrisChan888) [reply src/components/DepositButton.tsx:27]: It's awsome!

File: src/components/DepositButton.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    const token = tokenInstance.get()
+    const pool = PoolInstance.get()
+    const amount = utils.parseUnits(props.amount, TOKENS[props.symbol].decimals)
+    if (signer && pool && token) {
+      const signerAddress = await signer.getAddress()
+      const remainingAllowance = await token.allowance(
+        signerAddress,
+        PoolInstance.getCurrentAddress(),
+      )
+      if (remainingAllowance > amount) {
```
Comment (whisidev) [inline src/components/DepositButton.tsx:34]: >=

**Review by whisidev — APPROVED** [review]
LGTM. except the minor issue: >=

**Pull request merged**